File: server-pedidos-java/src/main/java/br/edu/utfpr/serverpedidosjava/util/validator/CEPValidator.java
package br.edu.utfpr.serverpedidosjava.util.validator;

import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;

public class CEPValidator implements ConstraintValidator<CEP, String> {

    @Override
    public boolean isValid(String value, ConstraintValidatorContext context) {
        return value == null || value.isEmpty() || value.matches("^\\d{5}\\-?\\d{3}$");
    }

}

File: server-pedidos/src/main/kotlin/br/edu/utfpr/serverpedidos/util/validator/CEPValidator.kt
package br.edu.utfpr.serverpedidos.util.validator

import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext

class CEPValidator : ConstraintValidator<CEP, String> {
    override fun isValid(value: String?, context: ConstraintValidatorContext?): Boolean {
        return value.isNullOrEmpty() || value.matches(Regex("^\\d{5}\\-?\\d{3}\$"))
    }
}
